test/build/create_build_test.go

```diff
@@ -0,0 +1,79 @@
+package build
+
+import (
+	"os"
+	"testing"
+
+	"github.com/bugsnag/bugsnag-cli/pkg/build"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSetBuilderName(t *testing.T) {
+	t.Log("Test setting builders name")
+	results, err := build.SetBuilderName("foobar")
+	if err != nil {
+		t.Error(err)
+	}
+	assert.Equal(t, "foobar", results, "They should be the same")
+
+	t.Log("Test not setting the builders name")
+	results, err = build.SetBuilderName("")
+	if err != nil {
+		t.Error(err)
+	}
+	assert.Equal(t, os.Getenv("USER"), results, "They should be the same")
+}
+
+func TestGettingRepoInfo(t *testing.T) {
+	t.Log("Test getting repo info map")
+	results, err := build.GetRepoInfo("", "", "")
+	if err != nil {
+		t.Error(err)
+	}
+	assert.Equal(t, map[string]string{
+		"repository": "https://github.com/bugsnag/bugsnag-cli",
+		"revision":   os.Getenv("GIT_SHA"),
```

Comment:
This is more general than this one test/assertion, but I'd instead avoid asserting on the revision auto-detected results in unit tests and rely on the features for this. As written, running `go test` directly results in broken tests since the makefile magic is not used.